mpldatacursor/pick_info.py

```diff
@@ -111,20 +112,15 @@ def line_props(event):
         return dict(x=xorig[i], y=yorig[i])
 
     # ax.step is actually implemented as a Line2D with a different drawstyle...
-    drawstyle = event.artist.get_drawstyle()
     xs_data = xorig[max(i - 1, 0) : i + 2]
     ys_data = yorig[max(i - 1, 0) : i + 2]
-    if drawstyle == "steps_pre":
-        xs_data = _interleave(xs_data, xs_data[:-1])
-        ys_data = _interleave(ys_data, ys_data[1:])
-    elif drawstyle == "steps_post":
-        xs_data = _interleave(xs_data, xs_data[1:])
-        ys_data = _interleave(ys_data, ys_data[:-1])
-    elif drawstyle == "steps_mid":
-        mid_xs = (xs_data[:-1] + xs_data[1:]) / 2
-        xs_data = _interleave(xs_data, np.column_stack([mid_xs, mid_xs]))
-        ys_data = _interleave(
-            ys_data, np.column_stack([ys_data[:-1], ys_data[1:]]))
+    drawstyle_conv = {
+        "_draw_lines": lambda xs, ys: (xs, ys),
+        "_draw_steps_pre": cbook.pts_to_prestep,
```

Comment:
@joferkington I can obviously restore the previous implementation, but perhaps that's a good time to pick a minimal version of matplotlib to support?  I see a few back-compatibility snippets here and there in the codebase.